File: backend/entities/movie.py
from typing import Optional
from pydantic import BaseModel

class Movie(BaseModel):
    id: int
    title: Optional[str] = None
    original_title: Optional[str] = None
    poster_path: Optional[str] = None
    overview: Optional[str] = None
    release_date: Optional[str] = None
    vote_average: Optional[float] = None
    vote_count: Optional[int] = None
    genres: Optional[list] = None

    class Config:
        orm_mode = True

File: src/entities/Movie.ts
export type Genre = { id: number; name: string };

export type Movie = {
  id: number;
  title?: string;
  original_title?: string;
  poster_path?: string | null;
  overview?: string;
  release_date?: string;
  vote_average?: number;
  vote_count?: number;
  genres?: Genre[];
};
